gcloud/orb.yaml

```diff
@@ -59,10 +59,36 @@ commands:
     parameters:
       cluster:
         type: string
+      kubectl_version:
+        default: "gcloud-latest"
+        description: |
+          kubectl version to be installed. Defaults to "gcloud-latest", ie. the version returned by installing as a gcloud component. You can specify the version string (eg. "v1.21.3") or "latest" to get a different version without going through gcloud components.
+        type: string
       zone:
         type: string
     steps:
-      - run: echo y | gcloud components install kubectl
+      - when:
+          condition:
+            equal: [ <<parameters.kubectl_version>>, "gcloud-latest" ]
+          steps:
+            - run: echo y | gcloud components install kubectl
+      - when:
+          condition:
+            equal: [ <<parameters.kubectl_version>>, "latest" ]
+          steps:
+            - run: curl -Lso/tmp/kubectl-version https://dl.k8s.io/release/stable.txt
+            - run: curl -Lo/tmp/kubectl "https://dl.k8s.io/release/$(cat /tmp/kubectl-version)/bin/linux/amd64/kubectl"
+            - run: install -o root -g root -m 0755 /tmp/kubectl /usr/local/bin/kubectl
```

Comment:
Interesting, is this a common pattern for installs? I mean, I see it matches what's here https://www.devopsschool.com/blog/how-to-install-and-configure-kubectl/ !

Reply:
Yeah, `install` is basically a unix builtin which is equivalent to `cp .. && chown .. && chmod ..`, for pretty much exactly this use-case.